### ISSUE TITLE
`@stylistic/declaration-bang-space-before` fix takes `!important` into an inline comment

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#116]
Found while fixing #100, by sweeping every rule that deletes a trailing whitespace run before writing. The same shape as #63, #87, #88 and #89, and a sibling of #113 and #114: an inline comment ends only with a line break, so whatever the fix pulls onto its line is commented out. Here it is `!important`, and the semicolon behind it.

### What minimal example or steps are needed to reproduce the bug?

```scss
.a {
  color: red // c
    !important;
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "customSyntax": "postcss-scss",
  "plugins": "@stylistic/stylelint-plugin",
  "rules": {
    "@stylistic/declaration-bang-space-before": "never"
  }
}
```

`postcss-less` behaves the same, and so does `always`.

### What did you expect to happen?

The line break in front of `!important` is what closes the comment, so neither option can be satisfied by touching that whitespace: `never` would have to delete the line break, and `always` would have to replace it with a single space. The fix should leave the declaration alone and let the warning stand.

### What actually happened?

```scss
.a {
  color: red // c!important;
}
```

With `always`, one space further along:

```scss
.a {
  color: red // c !important;
}
```

Either way `!important` and the semicolon are now inside the comment: the declaration loses its importance and ends nowhere. No warning survives the run, so `--fix` reports a clean pass on a file it has just broken, and a second run finds nothing left to report.

The cause is at [`lib/rules/declaration-bang-space-before/index.js:73-81`](https://github.com/stylelint-stylistic/stylelint-stylistic/blob/main/lib/rules/declaration-bang-space-before/index.js#L73-L81): `targetBefore.replace(/\s*$/u, ``)` takes the line break for an ordinary trailing whitespace run. The text it cuts into is `decl.raws.important` ([`:62`](https://github.com/stylelint-stylistic/stylelint-stylistic/blob/main/lib/rules/declaration-bang-space-before/index.js#L62)), which holds everything standing between the value and `!important` — a comment included. A block comment in the same position is safe, since nothing about it depends on a line break.

`declaration-bang-space-after` needs nothing: it writes on the far side of the `!`, where no comment of this kind can stand.

### Do you have a proposal to fix the bug?

The shape the cluster settled on: where the text the fix would cut into ends with an inline comment, report the problem and write nothing. `endsWithInlineComment` in `lib/utils/` already answers that question, and `targetBefore` is what has to be put to it, in both branches.
